app/integration/runtime/src/main/java/io/syndesis/integration/runtime/IntegrationRouteBuilder.java

```diff
@@ -287,23 +287,18 @@ private void configureFlow(Flow flow, final String flowIndex) throws URISyntaxEx
             }
         } else {
             // not in a pipeline so set the step id header to be sure it is there
-            definition = parent.setHeader(IntegrationLoggingConstants.STEP_ID, constant(stepId));
+            parent.setHeader(IntegrationLoggingConstants.STEP_ID, constant(stepId));
```

Comment:
`setHeader` returns a `ProcessorDefinition<?> `and it is true that in this case it returns the same instance but reassigning it to a definition is IMHO more idiomatic as the underlying code is the free to return a different instance.